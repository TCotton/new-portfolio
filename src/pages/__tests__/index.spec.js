import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from "gatsby"
import Index from "../index"

beforeEach(() => {
	StaticQuery.mockImplementationOnce(({ render }) =>
		render({
			site: {
				siteMetadata: {
					description: 'whatever',
					title: `The portfolio and blog of web developer Andy Walpole`,
					author: `Andy Walpole`,
				},
			},
		})
	)
})

describe.skip("Index", () => {
	it("renders correctly", () => {
		const data = {
			site: {
				siteMetadata: {
					description: "whatever",
					author: "Andy Walpole",
					title: `The portfolio and blog of web developer Andy Walpole`
				},
			},
		}

		const tree = renderer.create(<Index data={data} />).toJSON();
		expect(tree).toMatchSnapshot();
	})
})
