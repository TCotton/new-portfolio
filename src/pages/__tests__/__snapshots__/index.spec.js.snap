// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
Array [
  <header>
    <div>
      <h1>
        <a
          href="/"
        >
          The portfolio and blog of web developer Andy Walpole
        </a>
      </h1>
    </div>
  </header>,
  <div>
    <main>
      <h1>
        Hi people
      </h1>
      <p>
        Welcome to your new Gatsby site.
      </p>
      <p>
        Now go build something great.
      </p>
      <div
        style={
          Object {
            "marginBottom": "1.45rem",
            "maxWidth": "300px",
          }
        }
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "paddingBottom": "100%",
                "width": "100%",
              }
            }
          />
          <img
            alt=""
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAACYklEQVQ4y42Uy28SQRjA+dM8efDmwYN6qF6qiSY+Y/WgQRMibY00TaWNBSRSCraYQtHl/bR0KyxQWCgWWAqU8izl/Sq7rLNsRHlVJpvJtzPfb77nDIOcZHSoqZSrp4+KtXIziubaLRysMCZiCYqOoVnhjNEi8RcztdxxeTzc6VBfT+5O2Vhpb+vw4wMdZ0ppWvP9xzLeJoDNThf2W+Nz1+XzNxQubSToSKKW+BDc+WOnkshhSVgeCiGpViZMEg1oxc26Knt+ae3bEtJTZwzE1kXLccG0+sOOlrcvZXvsczPkITfsa20vwIKnhsh+BnjUarT74Gb13CY7KBVJMv3z4N1NszQYsMWM62HNrCis/GxXn0iYls23uz5LPBcv0bH8hUH2XRoM85ySXv7JBtO87jMIvWq+H5GoYIHCLA1ZxD6Qap3Ak8IKfW7TJ50lK6uP9E6RgndHaODtCJ6Z5RyHfnE7j6gRbcKlCYNSt+rtETHTpUGgEP8FYmdNqd/Mo7aiVWTfuH2L9xASvfxxlqr01EYkrJszvNkgW9bH0OuFr+99m+y9IOeyU6zIp/Hubp/yMEztlzFPwOhdvq+nIoS1JNn4t2sugCmVsDvPe2KKolnZLCxhOcAKQRDDXTQaVi46lqYhIBwHTrl3oWqhMRDtaJge37lOBMKo4tfbqhVX0J7snTsWps8uZWuoOQY6CcjpSIF55UvmqNgr5wUwtV1IVdnXtnSfPEB2qjDNqnvczRl0m+j6Jn5lXb6nAQJqinmN0ZEBj03YLzghY8PnTRz80o/GRJZpOLCb0PM9BN7pvUEjx28V00WUg9jIVwAAAABJRU5ErkJggg=="
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "transitionDelay": "0.25s",
                "width": "100%",
              }
            }
          />
          <picture>
            <img
              alt=""
              onLoad={[Function]}
              sizes="(max-width: 300px) 100vw, 300px"
              src="/static/6d91c86c0fde632ba4cd01062fd9ccfa/fbe2f/gatsby-astronaut.png"
              srcSet="/static/6d91c86c0fde632ba4cd01062fd9ccfa/e1fed/gatsby-astronaut.png 75w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/08283/gatsby-astronaut.png 150w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/fbe2f/gatsby-astronaut.png 300w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/59257/gatsby-astronaut.png 450w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/26d9e/gatsby-astronaut.png 600w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a3fa0/gatsby-astronaut.png 800w"
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 0.5s",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><img sizes=\\"(max-width: 300px) 100vw, 300px\\" srcset=\\"/static/6d91c86c0fde632ba4cd01062fd9ccfa/e1fed/gatsby-astronaut.png 75w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/08283/gatsby-astronaut.png 150w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/fbe2f/gatsby-astronaut.png 300w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/59257/gatsby-astronaut.png 450w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/26d9e/gatsby-astronaut.png 600w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a3fa0/gatsby-astronaut.png 800w\\" src=\\"/static/6d91c86c0fde632ba4cd01062fd9ccfa/fbe2f/gatsby-astronaut.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </div>
      <a
        href="/page-2/"
      >
        Go to page 2
      </a>
      <div>
        <p>
          Count: 
          0
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Increment
        </button>
      </div>
      Days working as a full-time web developer
      Cups of coffee before I start work in the morning
      Industry RSS feeds I subscribe to using NewsBlur
      Minutes before I check my Twitter feed after I wake up
      Unit test coverage of this site using Karma and Jasmine
    </main>
    <div>
      container
    </div>
    <footer
      className="footer"
    >
      <div
        className="footer-top"
      >
        <a
          href="https://github.com/TCotton/portfolio"
        >
          <pre
            className="code-preview"
          >
            /**
 						* This entire site is coded in AngularJS and React. Inspect the code on GitHub: https://github.com/TCotton/portfolio
 						*/
 						 angular.module('portfolioApp', ['portfolioApp.blogAdminService','portfolioApp.blogAdminController','portfolioApp.blogCommentsDirective','portfolioApp.blogCommentsService','portfolioApp.blogPagesController','portfolioApp.blogPagesDirective','portfolioApp.blogPagesService','portfolioApp.blogPagesFilter','portfolioApp.blogSidebarController','portfolioApp.contractController','portfolioApp.contractService','portfolioApp.footerController','portfolioApp.footerService','portfolioApp.homepageDirective','portfolioApp.miscDirective','portfolioApp.sideProjectsController','portfolioApp.angularReact','portfolioApp.sitemapController','portfolioApp.wordProjectsController','AppConstants','ngResource','ngSanitize','ngRoute','ngAria','HashBangURLs','portfolioAppConfig','jmdobry.angular-cache','underscore','momentLibrary','helperFunctions','requestTimeout','react','detectLocalStorage','appTemplates'
]).config(function($routeProvider, $httpProvider) {

  $httpProvider.defaults.useXDomain = true;
  delete $httpProvider.defaults.headers.common['X-Requested-With'];

  $routeProvider
    .when('/', {
      templateUrl: 'homepage/main.html',
      title: 'The portfolio and blog of web developer Andy Walpole'
    })
    .when('/work-projects', {
      templateUrl: 'work-projects/my_work.html',
      title: 'Noteworthy work projects from the past 18 months',
      controller: 'WorkPageCtrl as WorkProjPageCtrl'
    })
    .when('/work-projects/:workPage', {
      templateUrl: 'work-projects/work_page.html',
      controller: 'WorkPageCtrl as WorkProjPageCtrl'
    })
 						
          </pre>
        </a>
      </div>
      <div
        className="footer-bottom"
      >
        <p
          className="footer-copyright"
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&copy; Andy Walpole / ",
              }
            }
          />
          <a
            href="/sitemap"
          >
            Sitemap
          </a>
        </p>
      </div>
    </footer>
  </div>,
]
`;
